subgraphs/aave-forks/protocols/aave-v3/src/mapping.ts

```diff
@@ -0,0 +1,832 @@
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+
+  // Set reserveFactor if not set, as setReserveFactor may be never called
+  const market = getOrCreateMarket(marketId, protocolData);
+  if (market.reserveFactor.equals(BIGDECIMAL_ZERO)) {
+    // see https://github.com/aave/aave-v3-core/blob/1e46f1cbb7ace08995cb4c8fa4e4ece96a243be3/contracts/protocol/libraries/configuration/ReserveConfiguration.sol#L377
+    // for how to decode configuration data to get reserve factor
+    const reserveFactorMask =
+      "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF0000FFFFFFFFFFFFFFFF";
+    const reserveFactorStartBitPosition = 64 as u8;
+
+    const maskArray = new Uint8Array(32);
+    maskArray.set(Bytes.fromHexString(reserveFactorMask));
+    // BITWISE NOT
+    for (let i = 0; i < maskArray.length; i++) {
+      maskArray[i] = ~maskArray[i];
+    }
+    // reverse for little endian
+    const reserveFactorMaskBigInt = BigInt.fromUnsignedBytes(
+      Bytes.fromUint8Array(maskArray.reverse())
+    );
+
+    const pool = LendingPoolContract.bind(event.address);
+    const poolConfigData = pool.getConfiguration(event.params.reserve).data;
+    const reserveFactor = poolConfigData
+      .bitAnd(reserveFactorMaskBigInt)
+      .rightShift(reserveFactorStartBitPosition);
+
+    log.info("[handleBorrow]reserveFactor set to {}", [
+      reserveFactor.toString(),
+    ]);
+    market.reserveFactor = reserveFactor
+      .toBigDecimal()
+      .div(exponentToBigDecimal(INT_TWO));
+    market.save();
+  }
+
+  _handleBorrow(
+    event,
+    event.params.amount,
+    marketId,
+    protocolData,
+    event.params.onBehalfOf
+  );
+}
+
+export function handleRepay(event: Repay): void {
+  const marketId = getMarketIdFromToken(event.params.reserve);
+  if (!marketId) {
+    log.error("[handleRepay]Failed to find market for asset {}", [
+      event.params.reserve.toHexString(),
+    ]);
+    return;
+  }
+  _handleRepay(
+    event,
+    event.params.amount,
+    marketId,
+    protocolData,
+    event.params.user
+  );
+}
+
+export function handleLiquidationCall(event: LiquidationCall): void {
+  const collateralMarketId = getMarketIdFromToken(event.params.collateralAsset);
+  if (!collateralMarketId) {
+    log.error("[handleLiquidationCall]Failed to find market for asset {}", [
+      event.params.collateralAsset.toHexString(),
+    ]);
+    return;
+  }
+
+  const debtMarketId = getMarketIdFromToken(event.params.debtAsset);
+  if (!debtMarketId) {
+    log.error("[handleLiquidationCall]Failed to find market for asset {}", [
+      event.params.debtAsset.toHexString(),
+    ]);
+    return;
+  }
+
+  _handleLiquidate(
+    event,
+    event.params.liquidatedCollateralAmount,
+    collateralMarketId,
+    protocolData,
+    event.params.liquidator,
+    event.params.user,
+    debtMarketId
+  );
+}
+
+/* //TODO: needed?
+
+export function handleMintedToTreasury(event: MintedToTreasury): void {
```

Comment:
I would say we don't want to remove any metrics that people might be using on aave v3. We could add a new helper field `market._accruedToTreasury` and then include this function

Reply:
Forgot to remove this chunk. I believe we no longer need the handleMintedToTreasury in the aave-forks code. The handler is used to handle protocol-side revenue, and its handle in `_handleReserveDataUpdated` using reserveFactor.